apps/betterangels/src/app/(private-screens)/add-note/[noteId].tsx

```diff
@@ -210,10 +210,7 @@ export default function AddNote() {
         throw new Error(`Failed to update interaction: ${updateError}`);
       }
 
-      if (revertBeforeTimestamp) {
-        return router.replace('/');
-      }
-      setSubmitted(true);
+      return router.replace(`/client/${data?.note.client?.clientProfile?.pk}`);
```

Comment:
seems like `clientProfile.pk` should not be optional here?
and if it is then we should handle the case when it's missing

Reply:
iiuc, the reason it's optional is because, while all clientprofiles have a user, only some users have a clientprofile

Reply:
re: `getClientProfileUrl()`, sure thing. where should that live?

Reply:
sure. But my question is what happens when we navigate to `/client/undefined`?

Reply:
sounds important
```
  const getClientProfileUrl = (clientProfileId: string | undefined) =>
    clientProfileId ? `/client/${clientProfileId}` : '/';
```